syntax = "proto3";
package micro_app;




import "model/app_store/app_version.proto";

/*
ListAppStoreMicroApp请求
*/
message ListAppStoreMicroAppRequest {
message Query {
    /*
    按name搜索,%xx%:模糊匹配,xx%:前缀匹配
    */
    string name = 1 ;
    /*
    按id列表查询
    */
    repeated string id = 2 ;
}
        
message Sort {
    /*
    按name排序，1为升序，-1为降序
    */
    int32 name = 1 ;
}
        
    
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 page_size = 2 ;
    /*
    查询条件
    */
    Query query = 3 ;
    /*
    条件查询：按照字段正/倒序 :1表示升序, -1表示降序
    */
    Sort sort = 4 ;
}

/*
ListAppStoreMicroApp返回
*/
message ListAppStoreMicroAppResponse {
message List {
message Icons {
    /*
    图标url
    */
    string large = 1 ;
}
        
    
    /*
    当前版本
    */
    app_store.AppVersion currentVersion = 1 ;
    /*
    小产品名称
    */
    string name = 2 ;
    /*
    小产品id
    */
    string id = 3 ;
    /*
    图标url
    */
    Icons icons = 4 ;
    /*
    小产品概要介绍
    */
    string intro = 5 ;
    /*
    功能预览
    */
    repeated string preview = 6 ;
    /*
    功能详细介绍
    */
    string description = 7 ;
    /*
    首页路径
    */
    string homepage = 8 ;
}
        
    
    /*
    返回总数
    */
    int32 total = 1 ;
    /*
    页数
    */
    int32 page = 2 ;
    /*
    该页大小
    */
    int32 page_size = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListAppStoreMicroAppApi返回
*/
message ListAppStoreMicroAppResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    micro_app.ListAppStoreMicroAppResponse data = 4 ;
}

